### PR TITLE
Problem: Deploying release artifacts to github is a manual process

### PR DESCRIPTION
**Pull request by sappo**
Solution: Use travis to deploy these artifacts automatically.

The deployment is triggered by tagging on the zeromq/libzmq repository.
Of the many builds travis is checking only the default one with
libsodium and drafts disabled is used for deployment.

For now the results of `make distcheck` are deployed as well as their
md5 and sha1 hash sums. Further changes may upload a generated
Changelog as well.

### CODE REVIEW

File: ci_build.sh [hunk 1]
```diff
@@ -29,13 +30,12 @@ if [ $BUILD_TYPE == "default" ]; then
...
-        ./configure "${CONFIG_OPTS[@]}" &&
-        export DISTCHECK_CONFIGURE_FLAGS="${CONFIG_OPTS[@]}" &&
-        make VERBOSE=1 distcheck
-    ) || exit 1
+    # Build and check this project
+    ./autogen.sh
+    ./configure "${CONFIG_OPTS[@]}"
+    make -j5
+    make check
+    make install
```
Comment (bluca) [inline ci_build.sh]: Is this change necessary? Dist-check does the above and more (most importantly it checks that make dist works), so won't we lose coverage?

Comment (sappo) [inline ci_build.sh]: I moved the dist-check to ci_deploy.sh but if it's worth running it to check the build I happily revert it.

Comment (bluca) [inline ci_build.sh]: I think so, otherwise we won't notice if make dist is broken until we try to do a release, which is not nice :-)

**Comment by sappo** [comment]
@bluca seems like by pushing the update I deleted our conversation. The original ci_build behavior is restored now.

**Comment by bluca** [comment]
Yes the comments on outdated diffs are hidden. At least now you can click on it and they will reappear, it used to be that they disappeared forever.

Anyway, this is great stuff, thanks!

**Comment by sappo** [comment]
I'll create a temporary tag now, to check if everything works as expected.

**Comment by bluca** [comment]
It worked! Awesome :-)

**Comment by sappo** [comment]
I'd approximately 60 travis builds on my fork to get it to to that :-D

**Comment by hitstergtd** [comment]
@sappo, thanks! looks good. Also, it would be good to post snapshot releases of the merge builds, so that in the future if we need to narrow down a commit that caused weird performance issues, it would just be a matter of downloading a snapshot release and then running the appropriate scripts. This was suggested in one of the issues recently.

**Comment by bluca** [comment]
git is better for that kind of work via git bisect, doing it via manually downloading hundreds of tarballs would be the stuff of nightmares!

**Comment by hitstergtd** [comment]
It might help you narrow down to a set of commits, but you still have to build them. Having those commits prebuilt makes things quicker to compare.

**Comment by bluca** [comment]
They are not prebuilt. This is not shipping binaries - that would be useless (and wrong) on Linux. It's shipping distributable tarballs.

**Comment by hitstergtd** [comment]
Ah, ok.
